### PR TITLE
[JSON manual] json search function

### PR DESCRIPTION
**Pull request by radudoros**
http://jira.cubrid.org/browse/CBRD-23492

Add json_extract, ->, ->> manual entries.

### CODE REVIEW

File: en/sql/function/json_fn.rst [hunk 1]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+.. function:: -> (json_doc column, json path)
+
+  Alias operator for JSON_EXTRACT with 2 arguments and the json_doc argument being a column.
+  Raises an error if a json path is NULL.
+  Returns NULL if it applied on a NULL json_doc argument.
+
+.. code-block:: sql
+
+    CREATE TABLE tj (a json);
+    INSERT INTO tj values ('{"a":1}'), ('{"a":2}'), ('{"a":2}'), (NULL);
```
Comment (eseokoh) [inline en/sql/function/json_fn.rst]: You want `3` not redundant `2`.

File: en/sql/function/json_fn.rst [hunk 2]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
     ======================
       "["$.a[0]","$.c[0]","$.d.e[0]"]"
+
+JSON_EXTRACT
+===================================
+
+.. function:: JSON_EXTRACT (json_doc, json path [ , json path] ...)
+
+  Returns json elements from the json_doc, that are addressed by the given paths.
+  If json path argments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array.
```
Comment (eseokoh) [inline en/sql/function/json_fn.rst]:
```suggestion
  If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containing json path are gathered in a resulting json array. 
```

File: en/sql/function/json_fn.rst [hunk 3]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+.. function:: ->> (json_doc column, json path)
+
+  Alias for JSON_UNQUOTE(column->json path). Operator can be applied only on json_doc arguments that are columns.
+  Raises an error if a json path is NULL.
+  Returns NULL if it applied on a NULL json_doc argument.
+
+.. code-block:: sql
+
+    CREATE TABLE tj (a json);
+    INSERT INTO tj values ('{"a":1}'), ('{"a":2}'), ('{"a":2}'), (NULL);
```
Comment (eseokoh) [inline en/sql/function/json_fn.rst]: `3`?

File: en/sql/function/json_fn.rst [hunk 4]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+JSON_EXTRACT
+===================================
+
+.. function:: JSON_EXTRACT (json_doc, json path [ , json path] ...)
+
+  Returns json elements from the json_doc, that are addressed by the given paths.
+  If json path argments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array. 
+  A single json element is returned if no wildcards are used in the given json paths and a single element is found, otherwise the json elements found are wrapped in a json array.
+  Raises an error if a json path is NULL.  
+  Returns NULL if no elements are found or if json_doc arguments is invalid.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: Returns null of json_doc argument is invalid? Shouldn't be an error?

Please note you have a typo `if json_doc arguments is `

File: en/sql/function/json_fn.rst [hunk 5]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+      3
+      NULL
+
+->>
+===================================
+
+.. function:: ->> (json_doc column, json path)
+
+  Alias for JSON_UNQUOTE(column->json path). Operator can be applied only on json_doc arguments that are columns.
+  Raises an error if a json path is NULL.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  Raises an error if the json path is NULL or invalid.
```

File: en/sql/function/json_fn.rst [hunk 6]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+    ======================
+      "["a"]" -- '$.d.e[0]' is the only path matching the given argument path family - paths that start with '.d' and end with an array index
+
+->
+===================================
+
+.. function:: -> (json_doc column, json path)
+
+  Alias operator for JSON_EXTRACT with 2 arguments and the json_doc argument being a column.
+  Raises an error if a json path is NULL.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
  Raises an error if a json path is NULL or invalid.
```

File: en/sql/function/json_fn.rst [hunk 7]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
       json_search('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', 'all', 'a', null, '$**[*]')
     ======================
       "["$.a[0]","$.c[0]","$.d.e[0]"]"
+
+JSON_EXTRACT
+===================================
+
+.. function:: JSON_EXTRACT (json_doc, json path [ , json path] ...)
```
Comment (ribram) [inline en/sql/function/json_fn.rst]:
```suggestion
.. function:: JSON_EXTRACT (json_doc, json path [, json path] ...)
```

File: en/sql/function/json_fn.rst [hunk 8]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+
+  Returns json elements from the json_doc, that are addressed by the given paths.
+  If json path argments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array. 
+  A single json element is returned if no wildcards are used in the given json paths and a single element is found, otherwise the json elements found are wrapped in a json array.
+  Raises an error if a json path is NULL.
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: What happens if a path does not exist?

File: en/sql/function/json_fn.rst [hunk 9]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
+::
+
+      json_extract('{"a":["a","b"],"b":"a","c":["a"], "d":{"e":["a"]}}', '$d**[*]')
+    ======================
+      "["a"]" -- '$.d.e[0]' is the only path matching the given argument path family - paths that start with '.d' and end with an array index
+
+->
+===================================
+
+.. function:: -> (json_doc column, json path)
```
Comment (ribram) [inline en/sql/function/json_fn.rst]: is there a way to write this as json_doc -> json_path?

File: en/sql/function/json_fn.rst (hunk 8 above)
Comment (radudoros) [reply en/sql/function/json_fn.rst]: "Returns NULL if no elements are found"

File: en/sql/function/json_fn.rst [hunk 10]
```diff
@@ -369,3 +369,111 @@ Accepting any paths that contain json array indexes will filter out '$.b'
...
     ======================
       "["$.a[0]","$.c[0]","$.d.e[0]"]"
+
+JSON_EXTRACT
+===================================
+
+.. function:: JSON_EXTRACT (json_doc, json path [, json path] ...)
+
+  Returns json elements from the json_doc, that are addressed by the given paths.
+  If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containg json path are gathered in a resulting json array.
```
Comment (eseokoh) [inline en/sql/function/json_fn.rst]:
```suggestion
  If json path arguments contain wildcards, all elements that are addressed by a path compatible with the wildcards-containing json path are gathered in a resulting json array. 
```

**Pull request merged**